psh: sysexec wildcards feature doc (#8)
phoenix-rtos/phoenix-rtos-doc: utils/psh.md

# Phoenix Shell (psh)

> Phoenix Shell is a simple command interpreter alowing to use basic Phoenix-RTOS functions.

Reviewer: Maybe it marketing, but it may fit better:
```suggestion
Phoenix Shell is a compact program that enables you to command the Pheonix RTOS from a command line.
```

> Phoenix Shell is a simple command interpreter alowing to use basic Phoenix-RTOS functions.

Author: I think that second is more slick and will be added as you wrote it.

## Login

Psh can be run in login mode by launching it from name pshlogin. User credentials are read from /etc/passwd file.

Additional user (of name "defuser") credentials can be passed using the PSH_DEFUSRPWDHASH enviroment variable defined at compile time. PSH_DEFUSRPWDHASH should store only the hash of a password. Then one can log in using:
- username: defuser

> - password: (corresponding with givnen hash)

Reviewer: ```suggestion
- password: (corresponding with given hash)
```

------------

## Sysexec command whitelist

> There is a feature of predefining command avalable for use with sysexec command in psh. These commands can be defined in two ways: 

Reviewer: ```suggestion
There is a feature of predefining command available for use with sysexec command in psh. These commands can be defined in two ways:
```

- /etc/wldcrds file
- PSH_WILDCARDS enviroment variable at compile time

If storing commands in /etc/wldcrds file each complete sysexec command should be stored in separate line with line length not exceeding 79 characters:

    sysexec argA1 argA2 argA3
    sysexec argB1 argB2
    sysexec argC1 argC2 argC3

> If storing commands in PSH_WILDCARDS enviroment variable each command should be ended by ';' sign:

Reviewer: ```suggestion
If storing commands in PSH_WILDCARDS environment variable each command should be ended by ';' sign:
```

Phoenix Shell is a compact program that enables you to control the Phoenix-RTOS from the command line.
## Login

`psh` can be run in login mode by launching it from name `pshlogin`. User credentials are read from `/etc/passwd` file.

Additional user credentials can be passed using the `PSH_DEFUSRPWDHASH` environment variable defined at compile time. `PSH_DEFUSRPWDHASH` should store only one hash of a password. Then one can also log in using:
- username: defuser
- password: (corresponding with given hash)

------------

## Sysexec command white list

The built-in whitelist functionality provides the ability to predefine an available set of `sysexec` commands. These commands can be defined in two ways: 
- `/etc/whitelist` file
- `PSH_SYSEXECWL` environment variable at compile time

If storing commands in `/etc/whitelist` file each complete `sysexec` command should be stored in separate line with line length not exceeding 79 characters:

    sysexec argA1 argA2 argA3
    sysexec argB1 argB2
    sysexec argC1 argC2 argC3

If the commands are stored in the `PSH_SYSEXECWL` environment variable, each command should end with a semicolon (`;`), as in the example below:

```bash

> export PSH_SYSEXECWL="sysexec argD1 argD2 argD3;sysexec argE1 argE2 argE3;sysexec argF1 argF2;

Reviewer 2: missing ending `"` char. Also: maybe remove trailing `;` as it's no longer needed

Author: `"` added and `;` removed. Thanks

```

> if neither `/etc/whitelist` and `PSH_SYSEXECWL is defined then sysexec will not have any restrictions.

Reviewer 2: missing "\`" after the `PSH_SYSEXECWL`

Author: Missing "`" added. Thanks
